src/lib/common/sol-platform.c

```diff
@@ -206,7 +207,7 @@ sol_platform_get_board_name(void)
     return board_name;
 }
 
-SOL_API int
+SOL_API enum sol_platform_state
 sol_platform_get_state(void)
 {
     return sol_platform_impl_get_state();
```

Comment:
After a brief talk, its better dont change the type of this function, because enum type its not the best way.